modules/core/arrow-test/src/main/kotlin/arrow/test/laws/MonoidalLaws.kt

```diff
@@ -0,0 +1,46 @@
+package arrow.test.laws
+
+import arrow.Kind
+import arrow.core.Tuple2
+import arrow.typeclasses.Applicative
+import arrow.typeclasses.Eq
+import arrow.typeclasses.Monoidal
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.forAll
+
+object MonoidalLaws {
+
+  fun <F> laws(
+          MDAL: Monoidal<F>,
+          AP: Applicative<F>,
+          EQ: Eq<Kind<F,  Tuple2<Int, Int>>>,
+          BIJECTION: (Kind<F, Tuple2<Tuple2<Int, Int>, Int>>) -> (Kind<F, Tuple2<Int, Tuple2<Int, Int>>>),
+          ASSOCIATIVE_SEMIGROUPAL_EQ: Eq<Kind<F, Tuple2<Int, Tuple2<Int, Int>>>>
+  ): List<Law> =
+    SemigroupalLaws.laws(MDAL, AP::just, BIJECTION, ASSOCIATIVE_SEMIGROUPAL_EQ) + listOf(
+      Law("Monoidal Laws: Left identity") { MDAL.monoidalLeftIdentity(AP::just, EQ) },
+      Law("Monoidal Laws: Right identity") { MDAL.monoidalRightIdentity(AP::just, EQ) }
+    )
+
+  fun <F> laws(
+          MDAL: Monoidal<F>,
+          f: (Int) -> Kind<F, Int>,
+          EQ: Eq<Kind<F, Tuple2<Int, Int>>>,
+          BIJECTION: (Kind<F, Tuple2<Tuple2<Int, Int>, Int>>) -> (Kind<F, Tuple2<Int, Tuple2<Int, Int>>>),
+          ASSOCIATIVE_SEMIGROUPAL_EQ: Eq<Kind<F, Tuple2<Int, Tuple2<Int, Int>>>>
+  ): List<Law> =
+    SemigroupalLaws.laws(MDAL, f, BIJECTION, ASSOCIATIVE_SEMIGROUPAL_EQ) + listOf(
+      Law("Monoidal Laws: Left identity") { MDAL.monoidalLeftIdentity(f, EQ) },
+      Law("Monoidal Laws: Right identity") { MDAL.monoidalRightIdentity(f, EQ) }
+    )
+
+  private fun <F> Monoidal<F>.monoidalLeftIdentity(f: (Int) -> Kind<F, Int>, EQ: Eq<Kind<F, Tuple2<Int, Int>>>): Unit =
+    forAll(Gen.int().map(f)) { fa: Kind<F, Int> ->
+      identity<Int>().product(fa).equalUnderTheLaw(identity(), EQ)
```

Comment:
I thought this would be equal to `fa`

Reply:
Of course you are right, don't know why I implemented it like this. Thanks for pointing that out!

Reply:
I think I remember why I did it like this. For example, if we look at ListK:
`fa = {1, 2} `
`fb = {}`
The current outcome of `fa x fb` is `{}`
`{}` is also the `identity` element of `ListK`, so the outcome within a `Monoidal` should actually be `fa x fb = fa` and not  `fa x fb = {}`. This kinda confuses me 😅. It would be great if you could shed some light on this.

Reply:
>The current outcome of fa x fb is {}

It shouldn't, should it? AFAIK the outcome should be the same as with monoid. That is, the concatenation of both lists.

Reply:
I am understanding the `Semigroupal` type class as an abstraction over the cartesian product so the current behaviour is `fa x {} = {}`

Reply:
Ooooooh. Never mind me then.

Reply:
But then I still have to change the documentation, right?